src/game/gamemodes/Gamemode.js

```diff
@@ -58,22 +58,22 @@ class Gamemode {
     const { iconID } = playerObject;
     const idTag = playerObject.id;
 
-    this.game.resourceServer
-      .requestResources([{ name: iconData[iconID].name, path: iconData[iconID].img }])
-      .then(resources => {
-        const circle = new PlayerCircle(this.game, resources[iconData[iconID].name]);
-        const controller = new PlayerController(this.game, idTag);
-        circle.setController(controller);
-        const backgroundCol = Number.parseInt(playerObject.backgroundColor.substr(1), 16);
-        const iconCol = Number.parseInt(playerObject.iconColor.substr(1), 16);
+    return new Promise((resolve, reject) => {

```

Comment:
Since this is now a promise the `callback` parameter should be removed and the `init()` method must be updated to use the promise instead of a callback for the local player to spawn.

Reply:
The idea was to avoid the overhead of creating an entire game entity. I see now that this became a bit messy though. Will fix